### ISSUE TITLE
[console] Add hook/function finder?

### ISSUE THREAD

**Issue by darol100** [hechoendrupal/DrupalConsole#662]
Using Drupal Console as learning tool, it would be nice to have some kinda of tool that allows me to enter a hook from D7 and tell me what command I need to used, or how generate the same but in D8. This would help user that have been build d7 modules, help them to understand how to build new app on D8.

This is something that you guys are interested on adding to this project ?

**Comment by seanr** [comment]
This would indeed be awesome. :+1:

**Comment by jmolivas** [comment]
Sounds like a great idea @darol100.

How do you think this can be added to the project?

Is that information somewhere available that we can reuse?

For the learning part, since we need to add extra information on the generators. I was thinking probably adding a glossary of terms using YAML files and take advantage of the multilingual capabilities of console to consume the information.

We can try to use the approach to take advantage and build that find/search functionality only once.

Feel free to discuss this and add new ideas, we are planning to start working again on the learning feature in a near future.

**Comment by jmolivas** [comment]
Not sure how much of that work was already done at the "Drupal Module Upgrader"
https://www.drupal.org/project/drupalmoduleupgrader

**Comment by darol100** [comment]
@jmolivas ,

> Is that information somewhere available that we can reuse?

I think the drupalmoduleupgrader have a lot of this work done. However, this is going to be a different approach. I have use the drupalmoduleupgrader, which is mainly to help you migrate your D7 to D8 module. However, a hook/function finder in this project would help user that are writing new D8 modules. Because I do not think nobody would write a D7 module and then try to use drupalmoduleupgrader to migrate to D8.

> How do you think this can be added to the project?

I do not know yet because I have not even look at the code of this project. I do not have enough experience on Symfony to make a decision like that. However, as an user I can give you an idea of how we should address this problem.

I think we should have a new command call `find` and two parameters `func` and `hook`. If the user type `drupal find:hook hookName` this would provide which command needs to be run in order to created this on D8. Using the --learning (or a similar option) that would provide some extra information about how this hook/func should work. We might be able to get data from the core just like `phpstorm`  to make the docs available.

**Comment by darol100** [comment]
I think this will be an awesome feature but without an external data to provide this type information this seem to be a very heavy operation of for this project. For this reason I'm closing this issue.
